eng/common/templates/steps/source-build.yml

```diff
@@ -97,6 +97,16 @@ steps:
       /p:AssetManifestFileName=$assetManifestFileName
   displayName: Build
 
+- script: $(Build.SourcesDirectory)/test.sh
```

Comment:
[Changes are not allow](https://github.com/dotnet/source-build-reference-packages/tree/main/eng/common#readme) to the eng/common content.  It will get overridden by arcade dependency flows.

This is going to make this work significantly harder.

Reply:
Did not realize this... In that case, this might involve having to move away from the Arcade templates for the SB jobs. From my perspective, I did not see a way to add in the test step without modifying the arcade templates.

Reply:
It is plausible to make chances to the arcade templates in the arcade repo but they would have to be changes that would make sense for all repos.  For example, maybe it would make sense to provide a mechanism to invoke additional steps that are passed in.  It might not make sense to expose that throughout all of the templates but more so it you just wanted to reuses the steps template directly.

Reply:
This makes sense. I'll open an issue in arcade to explore some options.

Reply:
FYI - https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-repo.md#advanced-granular-templates in case you haven't seen this documentation.